### ISSUE TITLE
Fullcalendar year view does not make use of HTML headings for month labels - Blocker for WCAG (accessibility)

### ISSUE THREAD

**Issue by PhilCousineau** [fullcalendar/fullcalendar#7537]
### Checklist

- [X] I've already searched through [existing tickets](https://github.com/fullcalendar/fullcalendar/issues)
- [X] Other people will find this feature useful

### Is this feature for a specific connector (React/Angular/etc)?

Angular

### Feature Description

In the year view calendar, HTML headings (H2 or H3) should be used for for month labels, in order to fulfil WCAG success criterion 1.3.1 Info and Relationships:  https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html

![YearViewFullcalendarMonthHeader](https://github.com/fullcalendar/fullcalendar/assets/136345677/c8e7c4aa-aa9b-43f0-b7ef-b8a02c2f2293)

**Comment by tvankith** [comment]
@acerix Can i work on this?

**Comment by arshaw** [comment]
I attempted to implement this in similar fashion to the main title ([here](https://github.com/fullcalendar/fullcalendar/issues/6972#issuecomment-2579118274)), however, after putting the month list in a `<div role=list`, any h2/h3/h4/etc within became invalid according to my a11y audit tools. It seems that a labelledby is the appropriate solution in this case. Please let me know if you have another perspective.
